### PR TITLE
lancedb_addtion_to_embedchain

### PR DESCRIPTION
**Pull request by kaushal07wick**
## Description

LanceDB integration to Embedchain

I've tried to implement the functions according to my understanding.

Please note this is not complete yet. I'm putting this up for review and would love to get suggestions and code modifications.

Thanks

### CODE REVIEW

**Comment by kaushal07wick** [comment]
added some tests and config file,

I haven't tested them yet

I am little ambiguous on the `lancedb.py` file

File: embedchain/vectordb/lancedb.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        if config is None and ld_config is None:
+            self.config = LanceDBConfig()
+        else:
+            if not isinstance(config, LanceDBConfig):
+                raise TypeError(
+                    "Config is not a `lancedbconfig` instance. "
+                    "Please make sure the type is right and that you are passing an instance. "
+                )
+            self.config = config or ld_config
+        self.connection = LanceDb(self.config.LD_URI, **self.config.LD_EXTRA_PARAMS)
```
Comment (AyushExel) [inline embedchain/vectordb/lancedb.py]: How will this work? lancedb.connection should be called here ?

File: embedchain/config/vectordbs/LanceDBConfig.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+from embedchain.config.vectordbs.BaseVectorDbConfig import BaseVectorDbConfig
+from embedchain.helper.json_serializable import register_deserializable
+
+@register_deserializable
+class LanceDBConfig(BaseVectorDbConfig):
+    def __init__(
+        self,
+        table_name: Optional[str] = None,
+        dir: Optional[str] = None,
+        ld_uri: Union[str, List[str]] = None,
```
Comment (AyushExel) [inline embedchain/config/vectordbs/LanceDBConfig.py]: make this uri

File: embedchain/config/vectordbs/LanceDBConfig.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        **LD_EXTRA_PARAMS: Dict[str, any],
```
Comment (AyushExel) [inline embedchain/config/vectordbs/LanceDBConfig.py]: kwargs

File: embedchain/config/vectordbs/LanceDBConfig.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        :param table_name: Default name for the collection, defaults to None
+        :type table_name: Optional[str], optional
+        :param dir: Path to the database directory, where the database is stored, defaults to None
+        :type dir: Optional[str], optional
+        :param ld_url: lancedb uri to be used for connection, defaults to None
+        :type ld_url: Union[str, List[str]], optional
+        :param LD_EXTRA_PARAMS: extra params dict that can be passed to lancedb.
+        :type LD_EXTRA_PARAMS: Dict[str, Any], optional
+        """
+        self.LD_URI = ld_uri or os.environ.get("LANCEDB_URI")
```
Comment (AyushExel) [inline embedchain/config/vectordbs/LanceDBConfig.py]: `self.uri` and don't take env vars

File: embedchain/vectordb/lancedb.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+@register_deserializable
+class LanceDb(BaseVectorDB):
+    """
+    LanceDB as vector database
+    """
+
+    def __init__(
+        self,
+        config: Optional[LanceDBConfig] = None,
+        ld_config: Optional[LanceDBConfig] = None,  #Backwards compatibility
```
Comment (AyushExel) [inline embedchain/vectordb/lancedb.py]: remove this

File: embedchain/vectordb/lancedb.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        """
+        if config is None and ld_config is None:
+            self.config = LanceDBConfig()
+        else:
+            if not isinstance(config, LanceDBConfig):
+                raise TypeError(
+                    "Config is not a `lancedbconfig` instance. "
+                    "Please make sure the type is right and that you are passing an instance. "
+                )
+            self.config = config or ld_config
```
Comment (AyushExel) [inline embedchain/vectordb/lancedb.py]: remove this

File: embedchain/vectordb/lancedb.py (hunk 1 above)
Comment (AyushExel) [reply embedchain/vectordb/lancedb.py]: lancedb.connect

**Pull request merged**